Add `data-testid` for toggle button

montelaidev opened the pull request:
## **Description**

This PR adds the prop `dataTestId` to the `ToggleButton` for easier testing.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

montelaidev committed:
feat: add data-testid for toggle button

danroc commented on ui/pages/settings/experimental-tab/experimental-tab.component.js:
```diff
@@ -304,8 +304,9 @@ export default class ExperimentalTab extends PureComponent {
                   {t('addSnapAccountToggle')}
                 </Text>
-                <Box data-testid="add-snap-account-toggle">
+                <Box>
```
Is the box still needed? I think I added it just to use the `data-testid`

montelaidev replied:
We can remove it.

removed in this commit 580f7138f9641dc678515a831fd15fa167773d54

montelaidev committed:
fix: remove unused box

gauthierpetetin committed:
Github action to check issue body matches issues templates and add labels to issue when needed (#21184)

## **Description**

This PR introduces a new Github action, to achieve the following:
1. Check if issue matches an existing issue template
('general-issue.yml' or 'bug-report.yml')
2. Add an `INVALID-ISSUE-TEMPLATE` label in case issue doesn't match any
of the templates
3. Add an `external-contributor` label in case issue was created by
someone external to the MetaMask organisation
4. Add a `regression-prod-x.y.z` label when possible, when `x.y.z` value
can be extracted from 'bug-report.yml' template

We will be able to leverage `regression-prod-x.y.z` and
`external-contributor` labels to improve our bug triage process and our
metrics collection.

[Same PR for Mobile
repo](https://github.com/MetaMask/metamask-mobile/pull/7379)

## **Manual testing steps**

1. Go on this [test
repo](https://github.com/gauthierpetetin-test/repo_test/issues/new?assignees=&labels=type-bug&projects=&template=bug-report-extension.yml&title=%5BBug%5D%3A+)
2. Fill the bug report issue form
6. Make sure the release version you indicate in the form has x.y.z
format
7. Click on "Submit new issue"
8. Wait for ~30s and 2 new labels shall appear on the issue you just
created (`regression-prod-x.y.z` and `external-contributor`)
9. Edit the issue's description by indicating a different release
version
10. Wait for ~30s and the `regression-prod-x.y.z` shall be updated
accordingly
11. Edit the issue by deleting the whole description
12. Wait for ~30s and the `INVALID-ISSUE-TEMPLATE` shall be added
 
## **Screenshots/Recordings**

### **Before**

NA

### **After**

https://recordit.co/Sq7fvCwkr5

## **Related issues**

Fixes https://github.com/MetaMask/MetaMask-planning/issues/1409

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [x] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Brad Decker <bhdecker84@gmail.com>

benjisclowder committed:
adding new test scenarios (#21321)

## **Description**


_Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

Adding test scenarios for manual QA of the tokens flows.

## **Manual testing steps**

_1. Step1:_
_2. Step2:_
_3. ..._

## **Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the
before and after of your change._

### **Before**

_[screenshot]_

### **After**

_[screenshot]_

## **Related issues**

_Fixes #???_

https://github.com/MetaMask/metamask-extension/issues/20256 and
https://github.com/MetaMask/metamask-extension/issues/20255

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Peter <53189696+PeterYinusa@users.noreply.github.com>

montelaidev committed:
Update copy for account snaps (#21322)

## **Description**

This PR updates the copy of Account Snap related content.

## **Manual testing steps**

1. Click on Settings Icon
2. Click Experimental
3. See the Accounts Snaps section for the new copy

Part 2.
1. Click on the account picker
2. Click on `Add account or hardware wallet`
3. See the new copy for the `Add account Snap`
4. Click on `Add account Snap`
5. See the copy for the `Add Account Snap Page`

## **Screenshots/Recordings**

### **Before**


![image](https://github.com/MetaMask/metamask-extension/assets/96463427/5fa19e4e-cdf1-4597-ae7d-eda01224d0a5)

![image](https://github.com/MetaMask/metamask-extension/assets/96463427/c819bdec-e5cd-4ce4-b519-61e8fcaa3a75)

![image](https://github.com/MetaMask/metamask-extension/assets/96463427/9532ba13-1100-4591-9a4a-3b60b7423df8)

![image](https://github.com/MetaMask/metamask-extension/assets/96463427/6c2b939f-e555-4172-ab10-ccbefdc28b23)

![image](https://github.com/MetaMask/metamask-extension/assets/96463427/a3ccb929-864a-4d9c-8cf0-7745282ff56d)

![image](https://github.com/MetaMask/metamask-extension/assets/96463427/23e912bf-1a68-4b66-9327-790c12110097)

![image](https://github.com/MetaMask/metamask-extension/assets/96463427/252bff56-2b66-4a19-af7f-0381dc4ac83f)

![image](https://github.com/MetaMask/metamask-extension/assets/96463427/7c63a5ca-1a64-434d-9bdd-2414557cbb04)


### **After**
<img width="591" alt="Screenshot 2023-10-11 at 22 35 36"
src="https://github.com/MetaMask/metamask-extension/assets/96463427/34066a8d-90a9-4094-babb-64fe0ff0c401">
<img width="254" alt="Screenshot 2023-10-11 at 22 36 53"
src="https://github.com/MetaMask/metamask-extension/assets/96463427/f1cfd312-a047-42e7-b2ee-9bc37a63e003">

<img width="359" alt="Screenshot 2023-10-11 at 22 36 40"
src="https://github.com/MetaMask/metamask-extension/assets/96463427/290dcc2e-9051-4ff2-bf2c-7981a013b0f1">
<img width="501" alt="Screenshot 2023-10-11 at 22 39 54"
src="https://github.com/MetaMask/metamask-extension/assets/96463427/ade6e1d8-713a-4c28-ae85-71b2ecd6a9a0">



## **Related issues**

Resolves https://github.com/MetaMask/accounts-planning/issues/103

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Daniel Rocha <68558152+danroc@users.noreply.github.com>

HowardBraham committed:
Add Account Snap MetaMetrics data to transactions and signatures (#21244)

## **Description**

Closes MetaMask/accounts-planning#71
Paired with Consensys/segment-schema#122

## **Manual testing steps**

1. Build from this branch or download the Flask build from this PR
2. Turn on MetaMetrics
3. In the Extensions page, open "Inspect views background.html"
4. Go to the "Network" tab
5. Filter by the word `batch`
6. Look for Signature and Transaction MetaMetrics events that come
through when you perform actions
7. If you have access to Mixpanel, look at Mixpanel to make sure the
events come in

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

montelaidev committed:
fix: update snapshot

zone-live committed:
[MMI] Updates mmi packages to latest version (#21341)

## **Description**

We are bumping MMI packages due to a small bug fix with error handling
in our sdk.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

gantunesr committed:
Merge branch 'develop' into feat/add-data-testid-to-toggle
